### PR TITLE
net-proxy/trojan: version bump to 1.13.0

### PR DESCRIPTION
**Pull request by YangMame**

### CODE REVIEW

**Comment by a17r** [comment]
Please rebase your PR on top of current gentoo.git HEAD.

**Comment by YangMame** [comment]
sorry for long time no reply, just came back from business trip

File: net-proxy/trojan/trojan-1.13.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+LICENSE="GPL-3+"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~x86"
+IUSE="mysql test"
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	>=dev-libs/boost-1.66.0:=
+	>=dev-libs/openssl-1.0.2:=
```
Comment (juippis) [inline net-proxy/trojan/trojan-1.13.0.ebuild]: Use a specific slot here.

File: net-proxy/trojan/trojan-1.13.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python{3_5,3_6,3_7} )
+
+inherit cmake-utils python-any-r1 systemd
+
+DESCRIPTION="An unidentifiable mechanism that helps you bypass GFW"
+HOMEPAGE="https://github.com/trojan-gfw/${PN}"
```
Comment (juippis) [inline net-proxy/trojan/trojan-1.13.0.ebuild]: Plaintext the `HOMEPAGE`, this should result in an error nowadays.

File: net-proxy/trojan/trojan-1.13.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	>=dev-libs/boost-1.66.0:=
+	>=dev-libs/openssl-1.0.2:=
+	mysql? ( dev-db/mysql-connector-c:= )
+"
+DEPEND="
+	${RDEPEND}
+	test? ( net-misc/curl ${PYTHON_DEPS} )
```
Comment (juippis) [inline net-proxy/trojan/trojan-1.13.0.ebuild]: You could multi-line this.

File: net-proxy/trojan/trojan-1.13.0.ebuild (hunk 1 above)
Comment (YangMame) [reply net-proxy/trojan/trojan-1.13.0.ebuild]: what's this mean?

Comment (juippis) [reply net-proxy/trojan/trojan-1.13.0.ebuild]: `dev-libs/openssl:0=`

File: net-proxy/trojan/trojan-1.13.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+LICENSE="GPL-3+"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~x86"
+IUSE="mysql test"
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	>=dev-libs/boost-1.66.0:=
+	>=dev-libs/openssl:0=
```
Comment (juippis) [inline net-proxy/trojan/trojan-1.13.0.ebuild]: Dropping `>=` should help. You should run `repoman full -x` locally before pushing to see if it works or not...

**Comment by juippis** [comment]
```
>>> Source compiled.
>>> Test phase: net-proxy/trojan-1.13.0
>>> Working in BUILD_DIR: "/tmp/portage/net-proxy/trojan-1.13.0/work/trojan-1.13.0_build"
ctest -j 16 --test-load 16 -j1
Test project /tmp/portage/net-proxy/trojan-1.13.0/work/trojan-1.13.0_build
    Start 1: LinuxSmokeTest-basic
1/2 Test #1: LinuxSmokeTest-basic .............***Failed    0.00 sec
    Start 2: LinuxSmokeTest-fake-client
2/2 Test #2: LinuxSmokeTest-fake-client .......***Failed    0.00 sec

0% tests passed, 2 tests failed out of 2

Total Test time (real) =   0.01 sec

The following tests FAILED:
	  1 - LinuxSmokeTest-basic (Failed)
	  2 - LinuxSmokeTest-fake-client (Failed)
Errors while running CTest
 * ERROR: net-proxy/trojan-1.13.0::testworld failed (test phase):
```
Can you get it working?

**Comment by YangMame** [comment]
> ```
> >>> Source compiled.
> >>> Test phase: net-proxy/trojan-1.13.0
> >>> Working in BUILD_DIR: "/tmp/portage/net-proxy/trojan-1.13.0/work/trojan-1.13.0_build"
> ctest -j 16 --test-load 16 -j1
> Test project /tmp/portage/net-proxy/trojan-1.13.0/work/trojan-1.13.0_build
>     Start 1: LinuxSmokeTest-basic
> 1/2 Test #1: LinuxSmokeTest-basic .............***Failed    0.00 sec
>     Start 2: LinuxSmokeTest-fake-client
> 2/2 Test #2: LinuxSmokeTest-fake-client .......***Failed    0.00 sec
> 
> 0% tests passed, 2 tests failed out of 2
> 
> Total Test time (real) =   0.01 sec
> 
> The following tests FAILED:
> 	  1 - LinuxSmokeTest-basic (Failed)
> 	  2 - LinuxSmokeTest-fake-client (Failed)
> Errors while running CTest
>  * ERROR: net-proxy/trojan-1.13.0::testworld failed (test phase):
> ```
> 
> Can you get it working?

No..it needs local network(python http.server,you can check the shell scripts),sorry i don't know how to make is works.

**Pull request closed without merging**